app/services/forest_liana/smart_action_form_parser.rb

```diff
@@ -0,0 +1,36 @@
+module ForestLiana
+  class SmartActionFormParser
+    def self.extract_fields_and_layout(form)
+      fields = []
+      layout = []
+      form&.each do |element|
+        if element[:type] == 'Layout'
+          # TODO: validate layout element
+          if %w[Page Row].include?(element[:component])
+            extract = extract_fields_and_layout_for_component(element)
+            element[:component] = element[:component].camelize(:lower)
+            layout << element
+            fields.concat(extract[:fields])
+          else
+            element[:component] = element[:component].camelize(:lower)
+            layout << element
+          end
```

Comment:
I wonder if by any chance the line
`element[:component] = element[:component].camelize(:lower)`
could be put just above the if? 🤔 

but not sure because of the use of the method: `extract_fields_and_layout_for_component`

Reply:
If I put the line before the if the `extract_fields_and_layout_for_component` function will not work.

Reply:
I have updated `extract_fields_and_layout_for_component` and add a comment to explain why `Page` becomes `page`